src/modpack/files.nim

```diff
@@ -36,22 +45,43 @@ const
   manifestFile* = joinPath(packFolder, "manifest.json")
   outputFolder* = joinPath(projectFolder, ".out/")
 
-proc initManifestFile(projectId: int, fileId: int): ManifestFile =
+proc initManifestMetadata*(name: string, explicit: bool, dependencies: seq[int]): ManifestMetadata =
+  result.name = name
+  result.explicit = explicit
+  result.dependencies = dependencies
+
+proc initManifestFile*(projectId: int, fileId: int, metadata: ManifestMetadata): ManifestFile =
   ## create a new manifest fmod object.
   result.projectId = projectId
   result.fileId = fileId
+  result.metadata = metadata
 
 converter toManifestFile(json: JsonNode): ManifestFile =
   ## creates a ManifestFile from manifest json
   result.projectId = json["projectID"].getInt()
   result.fileId = json["fileID"].getInt()
+  if json{"__meta"} == nil:
+    let cfMod = waitFor(fetchMod(json["projectID"].getInt()))
+    let cfModFile = waitFor(fetchModFile(json["projectID"].getInt(), json["fileID"].getInt()))
+    result.metadata.name = cfMod.name
+    result.metadata.explicit = true
+    result.metadata.dependencies = cfModFile.dependencies
```

Comment:
I'll probably need a `__paxManifestVersion` or something like that in the future for proper migrations,
but this is fine for now.